File: src/main/java/leets/leenk/domain/feed/application/util/FeedDescriptionUtil.java
package leets.leenk.domain.feed.application.util;


public class FeedDescriptionUtil {
    public static String normalizeDescription(String description) {
        if (description == null) return null;
        String trimmed = description.replaceFirst("^(\r\n|\n)+", "");

        return trimmed.replaceAll("(\r\n|\n){2,}", "\n");
    }
}

File: src/main/kotlin/leets/leenk/domain/feed/application/util/FeedDescriptionUtil.kt
package leets.leenk.domain.feed.application.util

object FeedDescriptionUtil {
    @JvmStatic
    fun normalizeDescription(description: String?): String? {
        if (description == null) return null

        val trimmed = description.replaceFirst(Regex("^(\r\n|\n)+"), "")

        return trimmed.replace(Regex("(\r\n|\n){2,}"), "\n")
    }
}
